x-pack/solutions/observability/plugins/synthetics/server/synthetics_service/sync_global_params.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import { EncryptedSavedObjectsPluginStart } from '@kbn/encrypted-saved-objects-plugin/server/plugin';
+import { SavedObjectsServiceStart } from '@kbn/core/server';
+import { getPrivateLocations } from './get_private_locations';
+import { SyntheticsMonitorClient } from './synthetics_monitor/synthetics_monitor_client';
+
+export const syncSpaceGlobalParams = async ({
+  spaceId,
+  savedObjects,
+  logger,
+  syntheticsMonitorClient,
+  encryptedSavedObjects,
+}: {
+  spaceId: string;
+  savedObjects: SavedObjectsServiceStart;
+  logger: Logger;
+  syntheticsMonitorClient: SyntheticsMonitorClient;
+  encryptedSavedObjects: EncryptedSavedObjectsPluginStart;
+}) => {
+  try {
+    logger.debug(`Synching global parameters of space with id ${spaceId}`);
+    const savedObjectsClient = savedObjects.createInternalRepository();
+    const allPrivateLocations = await getPrivateLocations(savedObjectsClient);
+    await syntheticsMonitorClient.syncGlobalParams({
+      spaceId,
+      allPrivateLocations,
+      soClient: savedObjectsClient,
+      encryptedSavedObjects,
+    });
+  } catch (error) {
+    logger.error(`Sync of global parameters for space with id ${spaceId} failed: ${error.message}`);
+  }
```

Comment:
since you have try/catch here, how will the pRetry work where this function s called?

Reply:
You're right!

I added the pRetry later forgetting that this function was wrapped in a try/catch.

The idea of the try/catch was that the endpoints should not fail if the sync of global parameters fails.. or should they return an error in that case?

Reply:
I moved pRetry inside the function [here](https://github.com/elastic/kibana/pull/216197/commits/fe3fa2ad9fa14de6d26a11a82ada3fc1b70c3600).